tests/integration/test_task_pool.py

```diff
@@ -516,6 +516,36 @@ async def test_trigger_states(
         assert task.is_manual_submit == should_trigger
 
 
+async def test_trigger_queue(one, run, db_select, complete, log_filter):
+    """It should handle triggering tasks in the queued state.
+
+    It should not be possible to use "trigger" to add flows to a task that is
+    in the queued state. This is not a firm requirement, we could reconsider
+    this behaviour at a later date, however, it is tricky as at present the
+    task's entry in the task_outputs table already exists by this point.
```

Comment:
OK this is why you think its tricky. Actually the same thing happens with any flow merge, since it only happens if a flow runs into an existing `n=0` task.  We just add a new flow row in the DB. 

```console
$ sqlite3 ~/cylc-run/bug/runN/log/db "select * from task_outputs"
1|bar|[1]|{"submitted": "submitted", "started": "started"}
1|foo|[1]|{}                                                       # <--- pre merge, waiting
1|foo|[1, 2]|{"submitted": "submitted", "started": "started"}      # <--- post merge, running
```

Reply:
Isn't it a problem leaving the `[1]` task behind in the DB?

Reply:
How is it a problem?

On the plus side, you can see exactly what happened (flow merge when foo was waiting).

And, as I said, the same thing happens for normal (non manual trigger) flow merges.